pkg/archive/archive.go

```diff
@@ -405,6 +405,8 @@ func Tar(path string, compression Compression) (io.ReadCloser, error) {
 // paths are included in `options.IncludeFiles` (if non-nil) or not in `options.ExcludePatterns`.
 func TarWithOptions(srcPath string, options *TarOptions) (io.ReadCloser, error) {
 
+	srcPath = fixVolumePathPrefix(srcPath)
```

Comment:
In Windows, a volume path already has the "\\?\" before the volume ID, so if a volume path is used (which happens when dealing with read/write layers) the path remains unchanged.  For regular paths, you can prepend in front of the drive letter, making the path "\\?\C:\Foo\" which will work as intended.